spinalcordtoolbox/download.py

```diff
@@ -106,42 +106,80 @@ def unzip(compressed, dest_folder):
 
 def install_data(url, dest_folder):
     """
-    Download data from a URL and install in the appropriate folder. Deals with multiple mirrors, retry download,
-    check if data already exist.
-    :param url: string or list or strings. URL or list of URLs (if mirrors).
-    :param dest_folder:
-    :return:
+    Download a data bundle from a URL and install in the destination folder.
+
+    :param url: URL or sequence thereof (if mirrors).
+    :param dest_folder: destination folder
+    :return: None
+
+    Notes:
+
+    - Uses `download_data()` to retrieve the data.
+    - Uses `unzip()` to extract the bundle.
+    - If destination already exists, it is cleared first,
+      from files that would be in the bundle (not the rest).
     """
 
-    # Download data
     tmp_file = download_data(url)
 
-    # unzip
-    dest_tmp_folder = sct.utils.tmp_create()
-    unzip(tmp_file, dest_tmp_folder)
-    extracted_files_paths = []
-    # Get the name of the extracted files and directories
-    extracted_files = os.listdir(dest_tmp_folder)
-    for extracted_file in extracted_files:
-        extracted_files_paths.append(os.path.join(os.path.abspath(dest_tmp_folder), extracted_file))
+    extraction_folder = sct.utils.tmp_create()
+
+    unzip(tmp_file, extraction_folder)
+
+    # Identify whether we have a proper archive or a tarbomb
+    with os.scandir(extraction_folder) as it:
+        has_dir = False
+        nb_entries = 0
+        for entry in it:
+            if entry.name in ("__MACOSX",):
+                continue
+            nb_entries += 1
+            if entry.is_dir():
+                has_dir = True
+
+    if nb_entries == 1 and has_dir:
+        # tarball with single-directory -> go under
+        with os.scandir(extraction_folder) as it:
+            for entry in it:
+                if entry.name in ("__MACOSX",):
+                    continue
+                bundle_folder = entry.path
+    else:
+        # bomb scenario -> stay here
+        bundle_folder = extraction_folder
 
-    # Check if files and folder already exists
     logger.info("Destination folder: {}".format(dest_folder))
-    logger.info("Checking if folder already exists...")
-    for data_extracted_name in extracted_files:
-        fullpath_dest = os.path.join(dest_folder, data_extracted_name)
-        if os.path.isdir(fullpath_dest):
-            logger.warning("Folder {} already exists. Removing it...".format(data_extracted_name))
-            shutil.rmtree(fullpath_dest)
-
-    # Destination path
-    # for source_path in extracted_files_paths:
-        # Copy the content of source to destination (and create destination folder)
-    distutils.dir_util.copy_tree(dest_tmp_folder, dest_folder)
+
+    os.makedirs(dest_folder, exist_ok=True)
```

Comment:
this change of behaviour might become a problem. I do remember that in the past, we did not remove the folder entirely, and remaining legacy files were conflicting with SCT (e.g. functions doing a glob.glob()). I know this is not a proper way of coding, but this is what it is (at least for now), so if we want to make sure to not break anything, why not simply removing the folder if it already exists?

Reply:
Code was updated to remove the destination directory.